### PR TITLE
Retain LGTM through squashes for kubernetes/kubernetes repo

### PR DESCRIPTION
**Pull request by gdsoumya**
Signed-off-by: Soumya Ghosh Dastidar <gdsoumya@gmail.com>

Closes kubernetes/kubernetes#103096

### CODE REVIEW

File: config/prow/plugins.yaml
Comment (sftim) [inline config/prow/plugins.yaml:150]: nit:
```suggestion
  - kubernetes/kubernetes
  - kubernetes/website
```
(sorting alphabetically)

Comment (gdsoumya) [reply config/prow/plugins.yaml]: fixed

**Comment by chases2** [comment]
Ironically, my only nit would be to squash the two commits :)

/approve
/hold for nit

**Comment by cblecker** [comment]
/hold

Has this been brought up to contribex yet?

cc @nikhita

**Comment by MadhavJivrajani** [comment]
@cblecker, @BenTheElder brought it up yesterday, thanks!

**Comment by sftim** [comment]
LGTM, FWIW

**Review by BenTheElder — APPROVED** [review]
LGTM
/hold

**Comment by nikhita** [comment]
This was originally enabled in test-infra but disabled in https://github.com/kubernetes/test-infra/pull/14426. The only reason mentioned is this :grimacing: 

> Also disable the tree hash thing that doesn't seem to be useful

@BenTheElder since you had lgtm'd that PR, do you recall specific reasons why this isn't being used in test-infra today?

**Comment by nikhita** [comment]
For what it's worth, this config option will only retain lgtm when the existing commits in a PR are squashed but NOT when the PR is rebased because that will change the tree hash. This seems like a safe option so I'm not really opposed to enabling this but I'm still interested to know why test-infra disabled it.

**Comment by sftim** [comment]
Yep, if your rebase does anything except a squash against the same merge base then an existing LGTM still gets cancelled.

**Comment by BenTheElder** [comment]
Memory is hazy but I recall it wasn't working (in a harmless lgtm is still lost on squash kind of way) at one point and hadn't seen it work in test-infra. sftim reports it working in website, I'm happy to enable it in test-infra and kind first and do more testing if we need but I also trust sftim.

Perhaps we can reference an example of WAI?

**Comment by sftim** [comment]
I'll see if I can find a PR where we've seen it work.

**Comment by sftim** [comment]
That's not quite the same thing, but I added an empty commit and then squashed it out with no trouble.

**Comment by nikhita** [comment]
Lgtm from me, but would also ack from @cblecker.

/assign @cblecker 

We'll also need to send a notice to k-dev once this is rolled out.

**Comment by MadhavJivrajani** [comment]
/cc @cblecker 
Could you PTAL?

**Comment by sftim** [comment]
I have a feeling that the _initial_ implementation didn't work right, but it's been behaving as advertised in [k/website](https://github.com/kubernetes/website/) for some time.

**Comment by cblecker** [comment]
/test pull-test-infra-verify-lint
/lgtm

So I did a bunch of testing on this, and the plugin does work as expected. I tried to get around it in a bunch of ways and was unable to. I think this is relatively safe.

I know it definitely wasn't working as expected back in 2018/2019 when @matthyx originally wrote the code.. I don't know if @nikhita's cleanup (https://github.com/kubernetes/test-infra/pull/10579) helped it, although I can't see exactly why. The only other thing I can think of is GitHub fixed some sort of bug in the tree hash they were returning.

We still need a roll out plan with communications, but I have no technical objections to moving this forward.

**Comment by matthyx** [comment]
For sure @nikhita magic helped <3

**Comment by matthyx** [comment]
If we can confirm it's now a solid foundation, I'll check whether we can skip CI using the same mechanism...
Thanks for following up on this @cblecker

**Comment by sftim** [comment]
Anyone want to blog about this on https://k8s.dev/? (if so, there are folks who can help make that happen)

**Comment by dims** [comment]
/retest

@cblecker @BenTheElder @nikhita what do we do with this?

**Comment by BenTheElder** [comment]
I am +1 in concept. I don't have context on it working or not currently or what process / approvals we need. cc @mrbobbytables

**Comment by mrbobbytables** [comment]
I am +1 on the idea as well. We should get comms rolling and maybe time it after 1.24 code thaw? Thoughts?

**Comment by BenTheElder** [comment]
1.24 has been delayed a bit but thaw is coming ...

**Comment by dims** [comment]
@cblecker @BenTheElder keeps happening repeatedly ( see https://kubernetes.slack.com/archives/C1J0BPD2M/p1652238461088849 ) can we please land this?

**Comment by dims** [comment]
@nikhita @mrbobbytables ping, do we want to get this going now?

**Comment by BenTheElder** [comment]
I guess for 1.26, sigh ... :/

**Comment by dims** [comment]
@kaslin @palnabarun @markyjackson-taulia @nikhita @MadhavJivrajani let's do this for 1.26 please?

**Comment by sftim** [comment]
I'm willing to help folks who aren't sure what to do (I might not be sure either, but still willing).

**Comment by mrbobbytables** [comment]
/remove-lifecycle stale

ouch ... did not remember this was still open :x will fire a comms msg out next week

**Comment by matthyx** [comment]
Yeah, apologizes @mrbobbytables too much things on my plate...

**Comment by matthyx** [comment]
> Yeah, apologizes @mrbobbytables too much things on my plate...

Sorry I mixed up with https://github.com/kubernetes/kubernetes/issues/105988 please disregard my previous comment.

**Comment by mrbobbytables** [comment]
/hold cancel

Deadline for comments has passed, we're good to go :)

**Comment by sftim** [comment]
I will add:
/lgtm

**Pull request merged**
